Refactored waja to add referral during visit

nKataraia opened the pull request.

nKataraia committed:
kwasasa

nKataraia committed:
for now

gosso22 committed:
Implement save referrals in ec_referral and modify Referral Registration event.

nKataraia committed:
Added PNC Referral and Child Referral flows in PNC and Child home visits

nKataraia committed:
fixed action appearances and added swahili translation for referral

gosso22 commented on opensrp-chw/build.gradle:
```diff
@@ -249,14 +250,15 @@ dependencies {
         exclude group: 'com.android.support', module: 'appcompat-v7'
         exclude group: 'id.zelory', module: 'compressor'
         exclude group: 'com.rengwuxian.materialedittext', module: 'library'
+        exclude group: 'com.ibm.fhir', module: 'fhir-path'
```
@nKataraia - I think this has been fixed so we dont need to exclude fhir-path library and include the version below.

gosso22 commented on opensrp-chw/build.gradle:
```diff
@@ -249,14 +250,15 @@ dependencies {
 //    }
 
-    implementation('org.smartregister:opensrp-client-chw-referral:1.3.13-NACP-ALPHA-SNAPSHOT@aar') {
+    implementation('org.smartregister:opensrp-client-chw-referral:1.3.13-NACP-LOCAL-ALPHA-SNAPSHOT@aar') {
```
@nKataraia - did you make any changes to the referral library? If not probably just leave this to the version others are using locally.

gosso22 commented on opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java:
```diff
@@ -16,42 +18,53 @@
-            evaluateMalnutritionScreening(serviceWrapperMap);
-            evaluateProblemSolving();
+            this.serviceWrapperMap=serviceWrapperMap;
+            evaluateToddlerDanger();
```
@nKataraia - evaluating other action inside danger signs makes danger signs mandatory, for the cases where there is no danger signs screening then no modules show with this implementation. Can you check out this scenario?

gosso22 commented on opensrp-chw/src/test/java/org/smartregister/chw/Futa.java:
@nKataraia - maybe you should remove this

gosso22 commented on opensrp-chw/src/nacp/java/org/smartregister/chw/util/BoolEvaluator.java:
@nKataraia - are you using this?

gosso22 reviewed (changes requested):
@nKataraia - I have left you a couple of comments for this PR, overall this is an awesome implementation. It is fun to review :). I think you might need to look into Child and PNC. For child the danger signs is made to be a condition for other modules, so for the case of children who dont have danger signs screening (for example children older than 12 months) the visit doesnt show any module. For PNC, if you select danger signs, then go back and select none. it still shows refer facility module. I think this should disappear and show other modules.

Thanks for working on this.

nKataraia committed:
fixed the issue of Interractor appearing when they shouldn't

nKataraia committed:
- reverted back gradle to the original branch version
- deleted Futa.java
- Made the action in Child Interractor conditional when the child is a toddler

nKataraia committed:
corrected the age range for toddler based on recurrence service json file

nKataraia committed:
refactored a bit and added a more descriptive variable

nKataraia committed:
corrected calculation of who is a toddler

nKataraia committed:
selected only danger sign services for todler to be more specific

nKataraia committed:
pulled only the relevant elements out of service array, and fixed a bug in JsonQ

gosso22 committed:
Pass activity context to the interactor.

gosso22 committed:
Create referral event and referral tasks when saving ANC home visit

gosso22 committed:
Fix clearing health facility selection action when appropriate.

gosso22 committed:
Added utility function to process referral events and create tasks.

gosso22 committed:
Pass context to the interactor.

gosso22 committed:
Refactor to get referral problems from interactor and pass it to the processReferral method.

gosso22 committed:
Tag referral event.

gosso22 committed:
Refactor remove creation of referral events.

gosso22 committed:
Implement referral tasks and events creation for PNC and Neonatal child.

gosso22 committed:
Revert library version.

gosso22 committed:
Refactor ChildHomeVisit to save referral event and create referral tasks.

issyzac commented on opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/AncHomeVisitInteractorFlv.java:
```diff
@@ -100,6 +104,31 @@ private void evaluateDangerSigns(Map<String, List<VisitDetail>> details,
+
+        if (details != null) ChwAncJsonFormUtils.populateForm(jsonForm, details);
+
+        JSONObject referralProblem=FacilitySelectionActionHelper.copyReferralProblem(referralPayload,"danger_signs_present");
```
Why do we have this implementation inside the action helper?

issyzac committed:
Merge branch 'addo-linkage-action' into refactor-waja-add-referral-during-visit

issyzac committed:
:hammer: fix codacy

issyzac committed:
:hammer: fix codacy issues

issyzac committed:
:hammer: fix codacy
